### PR TITLE
Feat/notifications page

### PR DESCRIPTION
**Pull request by Parle648**
## Describe your changes

- The UI for all types of notifications has been created
- Notifications are divided into groups:
    1. Today
    2. Yesterday
    3. This Month
    4. This Year
- For today's notifications, the number of hours between the current time and the time the notification was received is displayed
- If there are unread messages, their count is displayed in the footer
- After viewing the notifications, the information about unread messages is updated

## Issue ticket code (and/or) and link

- [As a Driver, I want to receive notifications, so that I can stay updated on important information](https://codecraftersintership.atlassian.net/browse/SCRUM-197?atlOrigin=eyJpIjoiOTMyYWM3NzgxM2NkNDVlNzgyYzNkM2Q3NTY5MWI5N2EiLCJwIjoiaiJ9)

## Video documentation
https://www.loom.com/share/e8e8d797d9774d5ea2a9d296619d66ff?sid=f5311498-a496-43e0-b5bd-d93ebc70a82f

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/NotificationsPage/components/NotificationsBlock/components/BellNotification/index.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  timeDifference: string;
+}
+
+const BellNotification: FC<BellNotificationProps> = ({ timeDifference }) => {
+  return (
+    <Box sx={notificationContainer}>
+      <img src={bellNotification} alt='notificationIcon' />
+      <Box sx={textContainer}>
+        <Typography sx={notificationTitle}>
+          {t('You have received a note to the order')}
```
Comment (AlexeyBezrebriy) [inline src/pages/NotificationsPage/components/NotificationsBlock/components/BellNotification/index.tsx:25]: Not correct i18n

File: src/pages/NotificationsPage/components/NotificationsBlock/components/MapPinNotification/index.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+const MapPinNotification: FC<MapPinNotificationProps> = ({
+  routeId,
+  timeDifference,
+}) => {
+  return (
+    <Box sx={notificationContainer}>
+      <img src={notificationIcon} alt='notificationIcon' />
+      <Box sx={textContainer}>
+        <Typography sx={notificationTitle}>
+          {t('You have received new route ')}
```
Comment (AlexeyBezrebriy) [inline src/pages/NotificationsPage/components/NotificationsBlock/components/MapPinNotification/index.tsx:30]: Not correct i18n

**Review by AlexeyBezrebriy — APPROVED** [review]

File: src/pages/MainPage/components/Tab/styles.ts [hunk 3]
```diff
@@ -15,4 +15,37 @@ export const TabIcon = styled.div`
...
+  &.active img {
+    filter: brightness(1.2);
+  }
+`;
+
+export const Badge = styled.span`
+  position: absolute;
+  top: 0;
+  right: 0;
+  background-color: rgb(115, 103, 240);
```
Comment (oyakovytskyi) [inline src/pages/MainPage/components/Tab/styles.ts]: check 1.3.2 in this file

File: src/pages/NotificationsPage/components/NotificationsBlock/components/LuggageNotification/index.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <Box sx={notificationContainer}>
+      <img src={luggageNotification} alt='notificationIcon' />
+      <Box sx={textContainer}>
+        <Typography sx={notificationTitle}>
+          {t('The route ')}
+          <Link className='blackLink' to={`/routes/${routeId}`}>
+            #{routeId}
+          </Link>{' '}
+          {t('should be started at ')} {shouldBeStarted}
```
Comment (oyakovytskyi) [inline src/pages/NotificationsPage/components/NotificationsBlock/components/LuggageNotification/index.tsx:36]: {' '} - remove this ?
and white space at next line

File: src/pages/NotificationsPage/components/NotificationsBlock/useGetNotifications.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      dispatch(setUnreadMessages(null));
+    } catch (error) {
+      throw new Error('');
+    }
+  }, []);
+
+  const getNotifications = useCallback(async (): Promise<void> => {
+    try {
+      // await readAllNotifications();
+      const { data } = await axios.get(
```
Comment (oyakovytskyi) [inline src/pages/NotificationsPage/components/NotificationsBlock/useGetNotifications.ts:64]: no comments 1.1.10

File: src/pages/NotificationsPage/components/NotificationsBlock/useGetNotifications.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+      setNotifications(data);
+    } catch (error) {
+      throw new Error('');
+    }
+  }, [accessToken, user?.id]);
+
+  useEffect(() => {
+    getNotifications().catch();
+    readAllNotifications().catch();
+    // dispatch(setUnreadMessages(null));
```
Comment (oyakovytskyi) [inline src/pages/NotificationsPage/components/NotificationsBlock/useGetNotifications.ts]: no comments 1.1.10

File: src/pages/NotificationsPage/components/NotificationsBlock/useGetNotifications.ts (hunk 5 above)
Comment (Parle648) [reply src/pages/NotificationsPage/components/NotificationsBlock/useGetNotifications.ts:64]: done

File: src/pages/NotificationsPage/components/NotificationsBlock/useGetNotifications.ts (hunk 6 above)
Comment (Parle648) [reply src/pages/NotificationsPage/components/NotificationsBlock/useGetNotifications.ts]: done

File: src/pages/NotificationsPage/components/NotificationsBlock/components/LuggageNotification/index.tsx (hunk 4 above)
Comment (Parle648) [reply src/pages/NotificationsPage/components/NotificationsBlock/components/LuggageNotification/index.tsx:36]: it's prettier if I'm not mistaken

Comment (Parle648) [reply src/pages/NotificationsPage/components/NotificationsBlock/components/LuggageNotification/index.tsx:36]: done

File: src/pages/MainPage/components/Tab/styles.ts (hunk 3 above)
Comment (Parle648) [reply src/pages/MainPage/components/Tab/styles.ts]: fixed

**Review by asdelay — APPROVED** [review]

**Review by oyakovytskyi — APPROVED** [review]

File: src/utils/createIdString.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { ID_PREFIX_LENGTH } from '@/constants/numbers';
+
+export const createIdString = (id: string): string => {
```
Comment (lyudad) [inline src/utils/createIdString.ts:16]: добавь коментарий зачем эта функция нужна и/или пример входных-выходных параметров (тоже коментами над функцией)

Comment (Parle648) [reply src/utils/createIdString.ts:16]: done

**Pull request merged**